addons/stock/models/stock_move.py

```diff
@@ -179,11 +179,48 @@ def _default_group_id(self):
     next_serial = fields.Char('First SN')
     next_serial_count = fields.Integer('Number of SN')
 
+    will_be_late = fields.Boolean('Will be late', compute='_compute_will_be_late', default=False)
+
     @api.onchange('product_id', 'picking_type_id')
     def onchange_product(self):
         if self.product_id:
             self.description_picking = self.product_id._get_description(self.picking_type_id)
 
+    @api.depends('date_expected', 'move_orig_ids.date_expected', 'date', 'state')
+    def _compute_will_be_late(self):
+        for stock_move in self:
+            stock_move.will_be_late = stock_move.delay_alert and bool(stock_move._get_delay_time())
```

Comment:
i don't like it

i would prefer a normal stored boolean named `late` updated during write